src/libraries/System.Data.OleDb/src/OleDbConnection.COMWrappers.cs

```diff
@@ -0,0 +1,104 @@
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Data.OleDb
+{
+    public sealed partial class OleDbConnection
+    {
+        internal static unsafe Exception? ProcessResults(OleDbHResult hresult, OleDbConnection? connection, object? src)
+        {
+            if ((0 <= (int)hresult) && ((null == connection) || (null == connection.Events[EventInfoMessage])))
+            {
+                SafeNativeMethods.Wrapper.ClearErrorInfo();
+                return null;
+            }
+
+            // ErrorInfo object is to be checked regardless the hresult returned by the function called
+            Exception? e = null;
+            IntPtr pErrorInfo;
+            OleDbHResult hr = UnsafeNativeMethods.GetErrorInfo(0, &pErrorInfo);  // 0 - IErrorInfo exists, 1 - no IErrorInfo
+            if ((OleDbHResult.S_OK == hr) && (IntPtr.Zero != pErrorInfo))
+            {
+                UnsafeNativeMethods.IErrorInfo errorInfo = (UnsafeNativeMethods.IErrorInfo)OleDbComWrappers.Instance
+                    .GetOrCreateObjectForComInstance(pErrorInfo, CreateObjectFlags.UniqueInstance);;
+                try
+                {
+                    if (hresult < 0)
+                    {
+                        // UNDONE: if authentication failed - throw a unique exception object type
+                        //if (/*OLEDB_Error.DB_SEC_E_AUTH_FAILED*/unchecked((int)0x80040E4D) == hr) {
+                        //}
+                        //else if (/*OLEDB_Error.DB_E_CANCELED*/unchecked((int)0x80040E4E) == hr) {
+                        //}
+                        // else {
+                        e = OleDbException.CreateException(errorInfo, hresult, null);
+                        //}
+
+                        if (OleDbHResult.DB_E_OBJECTOPEN == hresult)
+                        {
+                            e = ADP.OpenReaderExists(e);
+                        }
+
+                        ResetState(connection);
+                    }
+                    else if (null != connection)
+                    {
+                        connection.OnInfoMessage(errorInfo, hresult);
+                    }
+                }
+                finally
+                {
+                    ((IDisposable)errorInfo).Dispose();
+                }
+            }
+            else if (0 < hresult)
+            {
+                // @devnote: OnInfoMessage with no ErrorInfo
+            }
+            else if ((int)hresult < 0)
+            {
+                e = ODB.NoErrorInformation((null != connection) ? connection.Provider : null, hresult, null); // OleDbException
+
+                ResetState(connection);
+            }
+            if (null != e)
+            {
+                ADP.TraceExceptionAsReturnValue(e);
+            }
+            return e;
+        }
+
+        internal void OnInfoMessage(UnsafeNativeMethods.IErrorInfo errorInfo, OleDbHResult errorCode)
```

Comment:
I don't see a difference in the code of `OnInfoMessage` between the COMWrappers file and the NoCOMWrappers file. Why do we need them split?